### DISCUSSION TITLE
tls error on a non tls server mariadb help

### DISCUSSION THREAD

**Discussion by sunmetis** [authelia/authelia#7485] [General]
Hi .. Trying to install authelia 4.3... but I have one last error reported from the log:
```

`authelia.service: Failed with result 'exit-code'.`

`juil. 03 09:51:04 debian-server systemd[1]: Started authelia.service - Authelia authentication and authorization server.`

`juil. 03 09:51:04 debian-server authelia[6205]: time="2024-07-03T15:51:04+02:00" level=info msg="Authelia v4.38.9 is starting"`

`juil. 03 09:51:04 debian-server authelia[6205]: time="2024-07-03T15:51:04+02:00" level=info msg="Log severity set to info"`

`juil. 03 09:51:14 debian-server authelia[6205]: time="2024-07-03T15:51:14+02:00" level=error msg="Error occurred running a startup check" error="error pinging database: TLS requested but server does not suppor>`

`juil. 03 09:51:24 debian-server authelia[6205]: time="2024-07-03T15:51:24+02:00" level=error msg="Error occurred running a startup check" error="failed to dial connection: EOF" provider=notification`

`juil. 03 09:51:24 debian-server authelia[6205]: time="2024-07-03T15:51:24+02:00" level=fatal msg="One or more providers had fatal failures performing startup checks, for more detail check the error level logs">`

`juil. 03 09:51:24 debian-server systemd[1]: authelia.service: Main process exited, code=exited, status=1/FAILURE`

`juil. 03 09:51:24 debian-server systemd[1]: authelia.service: Failed with result 'exit-code'.`
```

The mariadb database doen't need tls (on the synology) .

Here is the configuration.yml section concerned with "storage" configuration. I even disabled the tls in this file but still getting the same error.

```
`storage:`

`encryption_key: 'knaSCeum0rtX9iJcrFWhVhctIOGdRCdydwTePZPxDsujx6qAVdZlMnsIv1IzAz'`

`mysql:`

`address: 'tcp://10.0.100.11:3306'`

`database: 'authelia'`

`username: 'toto'`

`password: 'password'`

`timeout: '5s'`

`# tls: # Assurez-vous que cette ligne et le bloc suivant sont supprimés`

`# skip_verify: true`

```
Need you help .. THX

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

It looks like the issue is related to the TLS configuration for your MariaDB database. Even though you have disabled TLS in your `configuration.yml` file, the error message indicates that Authelia is still attempting to use TLS when connecting to the database.

Here are a few steps you can take to troubleshoot and resolve this issue:

1. **Ensure TLS is Completely Disabled**: Double-check that the TLS configuration is entirely removed or commented out in your `configuration.yml` file. It should look like this:

    ```yaml
    storage:
      encryption_key: 'knaSCeum0rtX9iJcrFWhVhctIOGdRCdydwTePZPxDsujx6qAVdZlMnsIv1IzAz'
      mysql:
        address: 'tcp://10.0.100.11:3306'
        database: 'authelia'
        username: 'toto'
        password: 'password'
        timeout: '5s'
        # tls: # Ensure this line and the block below are removed or commented out
        #   skip_verify: true
    ```

2. **Check for Environment Variables**: Ensure there are no environment variables that might be overriding the configuration file settings. Environment variables can sometimes take precedence over configuration file settings.

3. **Verify Database Connectivity**: Make sure that the MariaDB server is accessible from the Authelia server without TLS. You can use a MySQL client to test the connection:

    ```bash
    mysql -h 10.0.100.11 -u toto -p
    ```

4. **Review Logs for Additional Information**: Check the Authelia logs for any additional information that might help diagnose the issue. Look for any other errors or warnings that might provide more context.

5. **Update Authelia**: Ensure you are using the latest version of Authelia. Sometimes, bugs are fixed in newer releases that might resolve your issue.

If the issue persists, please provide more details about your environment and any additional configuration settings that might be relevant.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
Can you show your environment variables and upload your whole config as a [gist](https://gist.github.com/)?
